### PR TITLE
[CSPL-291]Adding common splunk-secrets object with unit test

### PR DESCRIPTION
**Pull request by akondur**

### CODE REVIEW

File: pkg/splunk/enterprise/util.go [hunk 1]
```diff
@@ -33,10 +34,69 @@ import (
...
+	secretTokenTypes := []string{"hec_token", "password", "pass4symmkey", "idxc_secret", "shc_secret"}
+
+	// Check if a K8S secrets object "splunk-secrets" exists in the namespace
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: commonSecretName}
+	err := client.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		// Not found, generate values for all types of tokens
+		secretData := make(map[string][]byte)
+		for _, tokenType := range secretTokenTypes {
+			secretData[tokenType] = splcommon.GenerateSecret(secretBytes, 24)
```
Comment (mikedickey) [inline pkg/splunk/enterprise/util.go:51]: Note that hec token is currently handled as a special case (formatting)

File: pkg/splunk/enterprise/util.go [hunk 2]
```diff
@@ -33,10 +34,69 @@ import (
...
+		current := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      commonSecretName,
+				Namespace: namespace,
+			},
+			Data: secretData,
+		}
+
+		// Create the secret
+		err = splctrl.CreateResource(client, &current)
```
Comment (mikedickey) [inline pkg/splunk/enterprise/util.go:63]: You could just return here. Also eliminates the need for and "else" block

File: pkg/splunk/enterprise/util.go [hunk 3]
```diff
@@ -33,10 +34,69 @@ import (
...
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Found, generate values for only missing types of tokens them
+		var updateNeeded bool = false
+		for _, tokenType := range secretTokenTypes {
+			if _, ok := current.Data[tokenType]; !ok {
+				// Value for token not found, generate
+				current.Data[tokenType] = splcommon.GenerateSecret(secretBytes, 24)
```
Comment (mikedickey) [inline pkg/splunk/enterprise/util.go:73]: note again special case for hec_token

Comment (akondur) [reply pkg/splunk/enterprise/util.go:83]: Fixed in the new commit 9a807f8cbe5f48b760a5123c35f5816d437f1752

File: pkg/splunk/enterprise/util.go (hunk 2 above)
Comment (akondur) [reply pkg/splunk/enterprise/util.go:67]: Fixed in the new commit 9a807f8cbe5f48b760a5123c35f5816d437f1752

File: pkg/splunk/enterprise/util.go (hunk 1 above)
Comment (akondur) [reply pkg/splunk/enterprise/util.go]: Fixed in the new commit 9a807f8cbe5f48b760a5123c35f5816d437f1752

**Comment by akondur** [comment]
Will reopen a new one.

**Pull request merged**